src/session-child.c

```diff
@@ -603,8 +603,13 @@ session_child_run (int argc, char **argv)
     if (x_authority)
     {
         gboolean drop_privileges = geteuid () == 0;
-        if (drop_privileges)
+        if (drop_privileges) {
+            if (x_authority_filename && chown (x_authority_filename, user_get_uid (user), user_get_gid (user)) < 0)
+                g_warning ("Failed to set ownership of user authority file: %s", strerror (errno));
             privileges_drop (user_get_uid (user), user_get_gid (user));
+            if (x_authority_filename && chmod (x_authority_filename, S_IRUSR | S_IWUSR) < 0)
```

Comment:
The chmod should be in x-authority.c I think - that's where it is set for newly created files.

Reply:
In the actual application process, an old file was accidentally modified with permissions, resulting in the inability to start a user session. This is done to remedy this abnormal situation.